crates/ibc/src/core/ics02_client/client_type.rs

```diff
@@ -1,36 +0,0 @@
-use crate::prelude::*;
-use core::fmt::{Display, Error as FmtError, Formatter};
-
-#[cfg_attr(
-    feature = "parity-scale-codec",
-    derive(
-        parity_scale_codec::Encode,
-        parity_scale_codec::Decode,
-        scale_info::TypeInfo
-    )
-)]
-#[cfg_attr(
-    feature = "borsh",
-    derive(borsh::BorshSerialize, borsh::BorshDeserialize)
-)]
-#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
-/// Type of the client, depending on the specific consensus algorithm.
-#[derive(Clone, Debug, PartialEq, Eq, PartialOrd, Ord, Hash)]
-pub struct ClientType(String);
```

Comment:
why does it make more sense for ClientType to be in ics24_host? It’s still used as a prefix to a ClientId; seems to make more sense to live under ics02_client?

Reply:
Since all the identifiers including the `ClientId` live in `ICS24_host`

Reply:
ICS-24 defines the concept of an identifier. `ClientType` is, well, the "type" of a client, which turns out is used to create an identifier; it is not an identifier itself. Basically, it is a "client" concept, which is also used to create identifiers. I think it makes more sense for it to live where it was

Reply:
https://github.com/cosmos/ibc-rs/pull/639/commits/94bfc5ef24506a2e6e16664c30d680b45c139a93